app/main/views/reset_password.py

```diff
@@ -139,3 +140,54 @@ def update_password(token):
                                email_address=email_address,
                                form=form,
                                token=token), 400
+
+
+@main.route('/change-password', methods=["GET", "POST"])
+@login_required
+def change_password():
+    form = ChangeOldPasswordForm()
+    if current_user.role == 'supplier':
+        dashboard_url = url_for('external.supplier_dashboard')
+    elif current_user.role == "buyer":
+        dashboard_url = url_for('external.buyer_dashboard')
+    elif current_user.role.startswith('admin'):
+        dashboard_url = url_for('external.admin_dashboard')
+    else:
+        dashboard_url = url_for('external.index')
```

Comment:
What's the `else` here? Anonymous users? Or is this a path that in theory should never be touched?

Reply:
Ah - I was going to refactor this! There's a similar bit of code in the user research view which I don't think checks for `admin` (which we do here, obviously).